### PR TITLE
LibC: Support `X` modifier for `scanf`

### PR DESCRIPTION
**Pull request by Quaker762**
This was currently crashing Half-Life because it was a considered an
"Unknown" specifier". We can use the same case statment, as the backend
to convert the number already supports upper/lower case input, hence
we get it for free :^)

### CODE REVIEW

File: Userland/Libraries/LibC/scanf.cpp [hunk 1]
```diff
@@ -488,6 +488,7 @@ extern "C" int vsscanf(const char* input, const char* format, va_list ap)
                 conversion_specifier = ConversionSpecifier::Unsigned;
                 break;
             case 'x':
+            case 'X':
```
Comment (alimpfard) [inline Userland/Libraries/LibC/scanf.cpp:491]: You know what they say
> Please add some tests for this.


I was also extremely surprised that `%X` behaves _literally_ the same as `%x` in glibc too, where it doesn't actually care about the casing.

Comment (alimpfard) [reply Userland/Libraries/LibC/scanf.cpp:491]: (fwiw, this isn't actually in POSIX: https://pubs.opengroup.org/onlinepubs/9699919799/functions/scanf.html)

Comment (Quaker762) [reply Userland/Libraries/LibC/scanf.cpp:491]: > You know what they say
> > Please add some tests for this.
> 
> 
> I was also extremely surprised that `%X` behaves _literally_ the same as `%x` in glibc too, where it doesn't actually care about the casing.

Same here honestly, I had to triple check to make sure I was getting Punk'd by POSIX

**Review by IdanHo — COMMENTED** [review]
Code looks fine, but about the commit message: `statment` => `statement`, also you should probably specify which existing case you're talking about (`'x'`)

**Comment by Quaker762** [comment]
> Code looks fine, but about the commit message: `statment` => `statement`, also you should probably specify which existing case you're talking about (`'x'`)

Done!

**Review by IdanHo — APPROVED** [review]
Thanks!

**Pull request merged**